cachito/workers/nexus.py

```diff
@@ -295,46 +295,98 @@ def search_components(**query_params):
     return items
 
 
-def upload_artifact(repo_name, repo_type, artifact_path):
+def upload_asset_only_component(repo_name, repo_type, component_path, to_nexus_hoster=True):
     """
-    Upload an artifact to the Nexus hosted repository.
+    Upload an asset only component to a Nexus hosted repository.
 
     :param str repo_name: the name of the Nexus hosted repository
     :param str repo_type: the type of the Nexus hosted repository (e.g. ``npm``)
-    :param str artifact_path: the path to the artifact to upload
+    :param str component_path: the path to the component to upload
+    :param bool to_nexus_hoster: Use the nexus hoster instance, if available
+    :raise CachitoError: if the upload fails
+    :raise ValueError: if uploading to an unsupported or non-asset-only repository type
+    """
+    NEXUS_ASSET_ONLY_UPLOAD_TYPES = ("pypi", "npm", "nuget", "rubygems")
+    if repo_type not in NEXUS_ASSET_ONLY_UPLOAD_TYPES:
+        raise ValueError(f"Type {repo_type!r} is not supported or requires additional params")
+
+    params = {"repository": repo_name}
+    with open(component_path, "rb") as component:
+        payload = {f"{repo_type}.asset": component.read()}
+
+    log.info("Uploading the component %r to the %r Nexus repository", component_path, repo_type)
+    try:
+        upload_component(params, payload, to_nexus_hoster)
+    except CachitoError:
+        log.exception("Failed to upload %r to the %r Nexus repository", component_path, repo_type)
+        raise
+
+
+def upload_raw_component(repo_name, directory, components, to_nexus_hoster=True):
+    """
+    Upload an asset only component to a Nexus hosted repository.
```

Comment:
The description should probably be different, this seems to be copied from the other upload function.

Reply:
Ha! You got me :see_no_evil: 
Fixed